### PR TITLE
#18344 Remove packages provided by framework

### LINKED ISSUE #18344: Remove polyfill packages for newer TFM
**Is your feature request related to a problem? Please describe.**
I want to minimise dependencies in my project by utilising framework dependencies wherever possible

**Describe the solution you'd like**
The Microsoft.CSharp package becomes conditional so only added for the older frameworks. Possibly other packages as well.

**Describe alternatives you've considered**
Accept the additional dependency

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by thompson-tomo**
### Issue link

closes #18344 

### Additional description

...Include details of the change made in this Pull Request or additional notes for the solution. Anything that can be useful for reviewers of this PR...

### Type of change

- [x] Optimization
- [x] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [ ] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [ ] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

File: src/Raven.Embedded/Raven.Embedded.csproj [hunk 1]
```diff
@@ -15,10 +15,10 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(OS)' == 'Windows_NT'">
-    <TargetFrameworks>netstandard2.0;net462</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net462;netcoreapp3.1</TargetFrameworks>
```
Comment (ppekrol) [inline src/Raven.Embedded/Raven.Embedded.csproj]: we removed netcoreapp3.1 because it is deprecated

Comment (thompson-tomo) [reply src/Raven.Embedded/Raven.Embedded.csproj]: Ok, would u accept net 5 or do u want it be net 6?

Comment (ppekrol) [reply src/Raven.Embedded/Raven.Embedded.csproj]: is there any need for this not to be netstandard2.0 and net462 only? Doesn't it cover everything anyway?

Comment (thompson-tomo) [reply src/Raven.Embedded/Raven.Embedded.csproj]: netstandard 2.0 requires the "System.Runtime.Loader" package where as net 3.1 & higher doesn't as it is provided natively the framework.

Comment (ppekrol) [reply src/Raven.Embedded/Raven.Embedded.csproj]: Ok, makes sense. I think lowest supported net would be best => `net6.0`

Comment (thompson-tomo) [reply src/Raven.Embedded/Raven.Embedded.csproj]: have switched to net6.0

**Comment by ppekrol** [comment]
@thompson-tomo Can you check compilation?

**Review by ppekrol — APPROVED** [review]

**Comment by arekpalinski** [comment]
> Hi!
> 
> Thank you for contributing into RavenDB project. Unfortunetely we detected that not all committers signed our CLA.
> 
> Following contributors are not in our database:
> 
>     * James Thompson ([thompson.tomo@outlook.com](mailto:thompson.tomo@outlook.com))
> 
> 
> Please visit http://ravendb.net/contributors/cla/sign to sign CLA.

@thompson-tomo Thanks for the contribution. Can you please sign CLA?

**Comment by thompson-tomo** [comment]
Have done it yesterday.

**Comment by arekpalinski** [comment]
Thanks. I missed that.

**Pull request merged**